scripts/qa_crowbarsetup.sh

```diff
@@ -3661,7 +3661,9 @@ function oncontroller_testsetup
     # wait for nova-manage to be successful
     wait_for 200 1 nova_services_up "Nova services to be up and running"
 
-    nova flavor-delete m1.smaller || :
+    if ! nova flavor-show m1.smaller 2> /dev/null ; then
```

Comment:
You are deleting `m1.smaller` if it *doesn't* exist. That's different than deleting it always and ignoring a failure. You should drop the negation.

Bash logic is always very confusing to me, so I ran this test on a SOC 7 deployment to see what it's doing. Since I didn't have `m1.smaller` I used `m1.small` instead:
```
# if ! nova flavor-show m1.small 2> /dev/null; then echo "deleting flavor"; else echo "not deleting"; fi
+----------------------------+----------+
| Property                   | Value    |
+----------------------------+----------+
| OS-FLV-DISABLED:disabled   | False    |
| OS-FLV-EXT-DATA:ephemeral  | 0        |
| disk                       | 20       |
| extra_specs                | {}       |
| id                         | 2        |
| name                       | m1.small |
| os-flavor-access:is_public | True     |
| ram                        | 2048     |
| rxtx_factor                | 1.0      |
| swap                       |          |
| vcpus                      | 1        |
+----------------------------+----------+
not deleting
```